web/lib/api/endpoint.ts

```diff
@@ -0,0 +1,85 @@
+import * as admin from 'firebase-admin'
+
+import { APIError } from 'common/api'
+import {
+  CORS_ORIGIN_MANIFOLD,
+  CORS_ORIGIN_LOCALHOST,
+  CORS_ORIGIN_VERCEL,
+} from 'common/envs/constants'
+import { removeUndefinedProps } from 'common/util/object'
+import { NextApiRequest, NextApiResponse } from 'next'
+import { applyCorsHeaders } from './cors'
+import { PrivateUser } from 'common/user'
+
+export const newEndpoint = (
+  fn: (req: NextApiRequest, userId: string) => Promise<Record<string, unknown>>
+) => {
+  return async (req: NextApiRequest, res: NextApiResponse) => {
+    try {
+      initAdmin()
+      await applyCorsHeaders(req, res, {
+        origin: [
+          CORS_ORIGIN_MANIFOLD,
+          CORS_ORIGIN_VERCEL,
+          CORS_ORIGIN_LOCALHOST,
+        ],
+        methods: 'POST',
```

Comment:
I think you need `allowedHeaders: 'Authorization'` here. But I don't know why it was working at all without it.

Reply:
Thanks. I haven't got as far as testing it yet haha.